3-extra/2-array-of-objects.js

```diff
@@ -12,6 +12,22 @@
 
 function getHighestRatedInEachGenre(books) {
     // TODO
+    const result = books.reduce((acc, cur) => {
+      const groupByGenre = cur.genre;
+      if (!acc[groupByGenre]) {
+        acc[groupByGenre] = [];
+      }
+      acc[groupByGenre].push(cur);
+      return acc;
+    }, {});
+    const genreName = Object.keys(result);
+    const arr = [];
+    for (let i = 0; i < genreName.length; i++) {
+      arr.push(
+        result[genreName[i]].sort((a, b) => b.rating - a.rating)[0].title
```

Comment:
This line could be broken up into several, so someone who isn't familiar with the code can follow what's happening